File: receiver/image_receiver.py
import cv2
import zmq
import numpy as np

# context = zmq.Context()
# footage_socket1 = context.socket(zmq.PULL)
# footage_socket1.bind("tcp://192.168.137.1:5555")

# footage_socket2 = context.socket(zmq.PULL)
# footage_socket2.bind("tcp://192.168.137.1:5556")

# print("here")
# while True:
#     try:
#         # Receive frame 1
#         frame_data1 = footage_socket1.recv()
#         nparr1 = np.frombuffer(frame_data1, np.uint8)
#         source1 = cv2.imdecode(nparr1, cv2.IMREAD_COLOR)

#         # Receive frame 2

#         # frame_data2 = footage_socket2.recv()
#         # nparr2 = np.frombuffer(frame_data2, np.uint8)
#         # source2 = cv2.imdecode(nparr2, cv2.IMREAD_COLOR)

#         # Display frames
#         cv2.imshow('Stream1', source1)

#         # cv2.imshow('Stream2', source2)
#         cv2.waitKey(10)

#         print("Success")

#     except zmq.error.ZMQError as e:
#         print(f"Exception: {e}")

# cv2.destroyAllWindows()


import tkinter as tk

def create_window(window_number):
    # 创建新窗口
    window = tk.Tk()
    window.title("Window {}".format(window_number))
    
    # 在窗口中添加组件或执行其他操作
    label = tk.Label(window, text="This is Window {}".format(window_number))
    label.pack()
    
    # 运行窗口的主循环
    window.mainloop()

# 设置N个窗口
N = 5

# 循环创建并运行N个窗口
for i in range(N):
    create_window(i+1)

File: mmdata/image_receiver.cpp
#include <opencv2/opencv.hpp>
#include<zmq.hpp>
#include <chrono>
#include <thread>

int main()
{
    
    zmq::context_t context(1);
    zmq::socket_t footage_socket1(context, ZMQ_PAIR);
    zmq::socket_t footage_socket2(context, ZMQ_PAIR);

    footage_socket1.bind("tcp://*:5555");
    footage_socket2.bind("tcp://*:5556");

   // 绑定到指定的地址和端口
    while(true){

        std::cout <<"listen" << std::endl;
        //接受 TCP传输过来的一帧视频图像
        zmq::message_t frame1;
        footage_socket1.recv(&frame1);

        zmq::message_t frame2;
        footage_socket2.recv(&frame2);

        //接受数据转化为opencv的图像数据
        std::string frameData1(static_cast<char*>(frame1.data()), frame1.size());
        std::vector<uchar> imgData1(frameData1.begin(), frameData1.end());
        cv::Mat source1 = cv::imdecode(cv::Mat(imgData1), cv::IMREAD_COLOR);


        std::string frameData2(static_cast<char*>(frame2.data()), frame2.size());
        std::vector<uchar> imgData2(frameData2.begin(), frameData2.end());
        cv::Mat source2 = cv::imdecode(cv::Mat(imgData2), cv::IMREAD_COLOR);


        //显示图像
        cv::imshow("Stream1", source1);
        cv::imshow("Stream2", source2);        
        cv::waitKey(1);

    }

    return 0;    
}